pkg/cni/token_watch.go

```diff
@@ -0,0 +1,103 @@
+package cni
+
+import (
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/fsnotify/fsnotify.v1"
+	"k8s.io/client-go/rest"
+)
+
+var serviceaccountDirectory string = "/var/run/secrets/kubernetes.io/serviceaccount/"
+var kubeconfigPath string = "/host/etc/cni/net.d/calico-kubeconfig"
```

Comment:
I'm concerned this is hard coded. But this is going to be dependent on how we mount the directory so maybe that is fine.